### PR TITLE
Replace stalebot with actions/stale action

### PR DESCRIPTION
**Pull request by hegyibalint**

### CODE REVIEW

**Comment by hegyibalint** [comment]
Note: needs to be merged for testing, as workflows are picked up from the default branch.

File: .github/workflows/stale-issue.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# GitHub action based on https://github.com/actions/stale
+
+name: 'Close stale issues and PRs'
```
Comment (wolfs) [inline .github/workflows/stale-issue.yml:3]: There is now a top-level `permissions` block which Github suggests adding to all workflows. Check out our other workflows.

File: .github/workflows/stale-issue.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v5
+        with:
+          # GLOBAL ------------------------------------------------------------
+          # !!! Debug is on until stabilized
+          debug-only: true
```
Comment (wolfs) [inline .github/workflows/stale-issue.yml:19]: Does that mean it won't actually change anything?

File: .github/workflows/stale-issue.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            Given the limited bandwidth of the team, it will be closed if no further activity occurs.
+            If you intend to work on this pull request, please ask the team to reopen the PR or push a new PR.
+            Thank you for your contributions.
+          days-before-pr-close: 21
+          close-pr-message: >
+            This issue has been automatically closed due to inactivity. If you can reproduce this on a
+            recent version of Gradle or if you have a good use case for this feature, please feel free
+            to to let know so we can reopen the issue. Please try to provide steps to reproduce, a quick
+            explanation of your use case or a high-quality pull request.
+          close-pr-reason: not_planned
```
Comment (wolfs) [inline .github/workflows/stale-issue.yml:55]: I don't think there is a `close-pr-reason`.

File: .github/stale.yml [hunk 4]
```diff
@@ -1,79 +0,0 @@
...
-    recent version of Gradle or if you have a good use case for this feature, please feel free
-    to to let know so we can reopen the issue. Please try to provide steps to reproduce, a quick
-    explanation of your use case or a high-quality pull request.
-
-# Only issues or pull requests with all of these labels are check if stale. Defaults to `[]` (disabled)
-#onlyLabels: []
-
-# Issues or Pull Requests with these labels will never be considered stale. Set to `[]` to disable
-exemptLabels:
-  - epic
```
Comment (wolfs) [inline .github/stale.yml:56]: This hasn't been migrated, see `exempt-issue-labels`

**Review by wolfs — COMMENTED** [review]
Looks mostly good, I left some comments.

File: .github/workflows/stale-issue.yml (hunk 1 above)
Comment (blindpirate) [reply .github/workflows/stale-issue.yml:3]: Actually there is a suggested permission list: https://github.com/actions/stale#recommended-permissions

File: .github/workflows/stale-issue.yml (hunk 2 above)
Comment (blindpirate) [reply .github/workflows/stale-issue.yml:19]: > Run the stale workflow as dry-run.
> No GitHub API calls that can alter your issues and pull requests will happen.
> Useful to debug or when you want to configure the stale workflow safely.

I wonder how we can know if it's work or not, by examining the action output log maybe?

Comment (hegyibalint) [reply .github/workflows/stale-issue.yml:19]: Yes, you can see the same output, just modifications are not applied.
See: https://github.com/gradle/gradle-private/actions/runs/2902491778

File: .github/workflows/stale-issue.yml (hunk 3 above)
Comment (hegyibalint) [reply .github/workflows/stale-issue.yml:55]: Correct, blindly copied it, and didn't realize it doesn't exists.

**Comment by hegyibalint** [comment]
Coordinating with the support team about handling community PRs.
After that, I think this is ready to be merged.

File: .github/stale.yml (hunk 4 above)
Comment (hegyibalint) [reply .github/stale.yml:56]: That's because I haven't found any epics.
Are those used anymore?

Comment (hegyibalint) [reply .github/stale.yml:56]: Hah! I've looked into private's labels... I can see we are using them here.
I will move this.

File: .github/workflows/stale-issue.yml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+          only-labels: 'from:contributor'
+
+          # ISSUES (deactivated) ----------------------------------------------
+          # This workflow should touch no issues, but actions/stale has no only-pr option.
+          # In order to deactivate this part, the following is done:
+          #  - Large stale time
+          #  - Only 'from:contributor' labels: this label should only appear oo PRs, and not on issues
+          #  - Exempt 'from:contributor' labels: forms a mutual exclusion with the only filter
+          #  - A bug message, so if the previous points stop working, it's obvious what happened
+          days-before-issue-stale: 3650
```
Comment (wolfs) [inline .github/workflows/stale-issue.yml:75]: Looks like you can use `-1` here to not mark issues as stale. Same for `days-before-issue-close`.

**Comment by hegyibalint** [comment]
I think this is ready.
Cross check stalebot config: https://github.com/gradle/gradle/blob/master/.github/stale.yml

Mentionable change is a separate workflow for the contributor PRs.

File: .github/workflows/stale-issue.yml (hunk 1 above)
Comment (hegyibalint) [reply .github/workflows/stale-issue.yml:3]: Added

**Review by wolfs — APPROVED** [review]
Looks good, I left one comment for a possible improvement.

File: .github/workflows/stale-issue.yml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+            **BUG!** This issue should not be marked stale by the "stale-community-pr" workflow.
+            Please report it to @bt-developer-productivity
+          days-before-issue-close: 3650
+          close-issue-message: >
+            **BUG!** This issue should not be closed by the "stale-community-pr" workflow.
+            Please report it to @bt-developer-productivity
+          close-issue-reason: not_planned
+
+          # PULL REQUESTS -----------------------------------------------------
+          days-before-pr-stale: 365
```
Comment (hegyibalint) [inline .github/workflows/stale-issue.yml:88]: Notified @gradle/bt-support about this.
This is a pretty high stale time, so we don't cause any trouble even if this is merged.

File: .github/workflows/stale-issue.yml (hunk 5 above)
Comment (hegyibalint) [reply .github/workflows/stale-issue.yml:75]: Hmm, I cannot read man pages

**Comment by hegyibalint** [comment]
@bot-gradle test and merge

**Pull request merged**